File: app/src/main/java/com/erif/contentloader/example/ActivityCombine.java
package com.erif.contentloader.example;

import static java.lang.Math.abs;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.erif.contentloader.LoaderContainer;
import com.erif.contentloader.R;
import com.erif.contentloader.helper.adapter.AdapterHorizontal;
import com.erif.contentloader.helper.adapter.AdapterVertical;
import com.erif.contentloader.helper.timer.Delay;
import com.google.android.material.imageview.ShapeableImageView;

import java.util.ArrayList;
import java.util.List;

public class ActivityCombine extends AppCompatActivity {

    private final AdapterHorizontal adapter = new AdapterHorizontal();
    private final AdapterVertical adapterVertical = new AdapterVertical();
    private final List<Integer> list = new ArrayList<>();
    private final List<Integer> listVertical = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_combine);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setTitle("Combined Loader");
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }

        LoaderContainer loader = findViewById(R.id.act_combined_loader);
        RelativeLayout image = findViewById(R.id.act_combined_layoutImage);
        loader.startAndHideContent(image, true);

        RecyclerView recyclerView = findViewById(R.id.act_combined_horizontal_recyclerView);
        setupList(recyclerView);

        LinearLayout horizontalContent = findViewById(R.id.act_combined_horizontal_contents);
        LoaderContainer loaderHorizontal = findViewById(R.id.content_loader_horizontal);
        loaderHorizontal.startAndHideContent(horizontalContent, true);

        new Delay(2, () -> loader.stopAndShowContent(image, true)).start();

        new Delay(2, () -> {
            adapter.setList(list);
            loaderHorizontal.stopAndShowContent(horizontalContent, true);
            recyclerView.scheduleLayoutAnimation();
        }).start();

        int sWidth = getResources().getDisplayMetrics().widthPixels;
        ShapeableImageView imgLeft = findViewById(R.id.act_combined_imgLeft);
        ViewGroup.MarginLayoutParams paramLeft = (ViewGroup.MarginLayoutParams) imgLeft.getLayoutParams();
        float snipped = sWidth * -1;
        float transX = snipped + (paramLeft.getMarginEnd() * 2);
        imgLeft.setTranslationX(transX);

        ShapeableImageView imgRight = findViewById(R.id.act_combined_imgRight);
        imgRight.setTranslationX(abs(transX));

        ShapeableImageView imgCenter = findViewById(R.id.act_combined_imgCenter);
        ViewGroup.MarginLayoutParams paramCenter = (ViewGroup.MarginLayoutParams) imgCenter.getLayoutParams();
        paramCenter.setMarginStart(paramCenter.getMarginStart() * 2);
        paramCenter.setMarginEnd(paramCenter.getMarginEnd() * 2);
        imgCenter.setLayoutParams(paramCenter);

        RecyclerView recyclerViewVertical = findViewById(R.id.act_combined_loader_vertical_recyclerView);
        setupListVertical(recyclerViewVertical);

        LoaderContainer loaderVertical = findViewById(R.id.act_combined_loader_vertical);
        loaderVertical.startAndHideContent(recyclerViewVertical, true);

        new Delay(2.2, () -> {
            adapterVertical.setList(listVertical);
            recyclerViewVertical.scheduleLayoutAnimation();
            new Delay(0.1, () ->
                    loaderVertical.stopAndShowContent(recyclerViewVertical, true)
            ).start();
        }).start();

    }

    private void setupList(RecyclerView recyclerView) {
        recyclerView.setAdapter(adapter);
        LinearLayoutManager manager = new LinearLayoutManager(this, RecyclerView.HORIZONTAL, false);
        recyclerView.setLayoutManager(manager);
        for (int i=0; i<20; i++) {
            list.add(i);
        }
    }

    private void setupListVertical(RecyclerView recyclerView) {
        recyclerView.setAdapter(adapterVertical);
        LinearLayoutManager manager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(manager);
        for (int i=0; i<20; i++) {
            listVertical.add(i);
        }
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/com/erif/contentloaderdemo/example/ActivityCombine.kt
package com.erif.contentloaderdemo.example

import android.os.Bundle
import android.view.MenuItem
import android.view.ViewGroup.MarginLayoutParams
import android.widget.LinearLayout
import android.widget.RelativeLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.erif.contentloader.LoaderContainer
import com.erif.contentloaderdemo.R
import com.erif.contentloaderdemo.helper.adapter.AdapterHorizontal
import com.erif.contentloaderdemo.helper.adapter.AdapterVertical
import com.erif.contentloaderdemo.helper.timer.Delay
import com.google.android.material.imageview.ShapeableImageView
import kotlin.math.abs

class ActivityCombine: AppCompatActivity() {

    private val adapter = AdapterHorizontal()
    private val adapterVertical = AdapterVertical()
    private val list: MutableList<Int> = ArrayList()
    private val listVertical: MutableList<Int> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_combine)
        supportActionBar?.let {
            it.title = "Combined Loader"
            it.setDisplayHomeAsUpEnabled(true)
        }

        val loader = findViewById<LoaderContainer>(R.id.act_combined_loader)
        val image = findViewById<RelativeLayout>(R.id.act_combined_layoutImage)
        loader.startAndHideContent(image, true)

        val recyclerView = findViewById<RecyclerView>(R.id.act_combined_horizontal_recyclerView)
        setupList(recyclerView)

        val horizontalContent = findViewById<LinearLayout>(R.id.act_combined_horizontal_contents)
        val loaderHorizontal = findViewById<LoaderContainer>(R.id.content_loader_horizontal)
        loaderHorizontal.startAndHideContent(horizontalContent, true)

        Delay(
            2
        ) { loader.stopAndShowContent(image, true) }.start()

        Delay(2) {
            adapter.setList(list)
            loaderHorizontal.stopAndShowContent(horizontalContent, true)
            recyclerView.scheduleLayoutAnimation()
        }.start()

        val sWidth = resources.displayMetrics.widthPixels
        val imgLeft = findViewById<ShapeableImageView>(R.id.act_combined_imgLeft)
        val paramLeft = imgLeft.layoutParams as MarginLayoutParams
        val snipped = (sWidth * -1).toFloat()
        val transX = snipped + paramLeft.marginEnd * 2
        imgLeft.translationX = transX

        val imgRight = findViewById<ShapeableImageView>(R.id.act_combined_imgRight)
        imgRight.translationX = abs(transX)

        val imgCenter = findViewById<ShapeableImageView>(R.id.act_combined_imgCenter)
        val paramCenter = imgCenter.layoutParams as MarginLayoutParams
        paramCenter.marginStart = paramCenter.marginStart * 2
        paramCenter.marginEnd = paramCenter.marginEnd * 2
        imgCenter.layoutParams = paramCenter

        val recyclerViewVertical =
            findViewById<RecyclerView>(R.id.act_combined_loader_vertical_recyclerView)
        setupListVertical(recyclerViewVertical)

        val loaderVertical = findViewById<LoaderContainer>(R.id.act_combined_loader_vertical)
        loaderVertical.startAndHideContent(recyclerViewVertical, true)

        Delay(2.2) {
            adapterVertical.setList(listVertical)
            recyclerViewVertical.scheduleLayoutAnimation()
            Delay(
                0.1
            ) { loaderVertical.stopAndShowContent(recyclerViewVertical, true) }.start()
        }.start()

    }

    private fun setupList(recyclerView: RecyclerView) {
        recyclerView.adapter = adapter
        val manager = LinearLayoutManager(this, RecyclerView.HORIZONTAL, false)
        recyclerView.layoutManager = manager
        for (i in 0..19) {
            list.add(i)
        }
    }

    private fun setupListVertical(recyclerView: RecyclerView) {
        recyclerView.adapter = adapterVertical
        val manager = LinearLayoutManager(this)
        recyclerView.layoutManager = manager
        for (i in 0..19) {
            listVertical.add(i)
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

}